### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by seantleonard**
## Why make this change?

- Explicitly adds `Azure.Identity` version 1.10.3 to relevant projects using `Microsoft.Data.SqlClient` so builds can pass.

## How was this tested?

- [x] Pass build pipeline which requires new package version.

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -49,4 +50,4 @@
     <PackageVersion Include="Verify.MsTest" Version="20.1.0" />
     <PackageVersion Include="Verify.DiffPlex" Version="2.2.1" />
   </ItemGroup>
-</Project>
+</Project>
```
Comment (Aniruddh25) [inline src/Directory.Packages.props:53]:
```suggestion
</Project>

```

File: src/Service/Azure.DataApiBuilder.Service.csproj [hunk 2]
```diff
@@ -48,6 +48,7 @@
 
 
     <ItemGroup>
+        <PackageReference Include="Azure.Identity" />
```
Comment (Aniruddh25) [inline src/Service/Azure.DataApiBuilder.Service.csproj:51]: This project doesnt depend on `Data.SqlClient`. Should we still include Azure.Identity explicitly here?

**Review by Aniruddh25 — APPROVED** [review]
LGTM, thank you for proactively fixing this and our pipeline!

**Review by aaronpowell — APPROVED** [review]

File: src/Service/Azure.DataApiBuilder.Service.csproj (hunk 2 above)
Comment (seantleonard) [reply src/Service/Azure.DataApiBuilder.Service.csproj:51]: Yes. Chatted offline.

**Pull request merged**